### PR TITLE
Added url quoting to X-Accel-Redirect and Location header

### PR DESCRIPTION
**Pull request by Proper-Job**
This fix is as unintrusive as I could make it.
All the tests pass on Python2.  As you know many tests fail on Python3.

I successfully tested this with a Nginx setup on PY2 and PY3.
I couldn't test this on Apache2 as I'm really unfamiliar with the setup.

Is there anything else you need?

### CODE REVIEW

**Comment by johnsensible** [comment]
Look this seems fine.

Need to get the Python setup on my laptop working again, so I can actually run the tests etc myself, but I can't see any reason why I won't merge this.

Thank you.

**Comment by johnsensible** [comment]
Ok, so I just ran this against Django 1.4 (which I know is quite old - though only fairly recently had support stopped) and get:

```
from django.utils.encoding import force_text, force_bytes
ImportError: cannot import name force_bytes
```

When was `force_bytes` added to Django?  We'll either need to make this work without `force_bytes` or change the `setup.py` etc to indicate which version of Django is supported.

Also I probably need to make it easier to run the tests - at the moment it's not so easy...

**Comment by Proper-Job** [comment]
@johnsensible Good catch.  I went for the smart variant of the encoding utils which in our case do the same thing as we're not using lazy object.  These were introduced in Django==1.4.2: https://github.com/django/django/blob/1.4.2/django/utils/encoding.py.

**Comment by Proper-Job** [comment]
@johnsensible ping

File: sendfile/backends/_internalredirect.py [hunk 1]
```diff
@@ -19,6 +19,6 @@ def _convert_file_to_url(filename):
         url.insert(1, head)
 
     # Python3 urllib.parse.quote accepts both unicode and bytes, while Python2 urllib.quote only accepts bytes.
-    # So force bytes for quoting and then go back to unicode.
-    url = [force_bytes(url_component) for url_component in url]
-    return force_text(quote(b'/'.join(url)))
+    # So use bytes for quoting and then go back to unicode.
+    url = [smart_bytes(url_component) for url_component in url]
+    return smart_text(quote(b'/'.join(url)))
```
Comment (johnsensible) [inline sendfile/backends/_internalredirect.py]: The only minor thing here is that the byte literal does tie this to Python 2.7+.  I guess that's not necessarily a bad thing and makes supporting Python 3 easier I guess.

**Comment by johnsensible** [comment]
Ok, sorry for the delay.  With work + kids it's pretty difficult to get the time for this.  Plus this account is still tied to my old work email, as I don't get notifications (I'd best sort that out).

Going to merge this, then I'll try and get a release together on PyPi over the weekend.

File: sendfile/backends/_internalredirect.py (hunk 1 above)
Comment (Proper-Job) [inline sendfile/backends/_internalredirect.py]: @johnsensible sorry for getting back to you so late.  I didn't introduce the byte literal.  I believe 2.7+ support is sensible as 2.6 is [officially](https://www.python.org/download/releases/2.6.9/) discontinued.

**Comment by Proper-Job** [comment]
@johnsensible Thank you and  I hear you ... kids + work redefines the definition of being slammed :)

**Comment by Proper-Job** [comment]
@johnsensible Are you going to push a new release to pypi?

**Comment by johnsensible** [comment]
Just trying to reset my pypi password now. Been a while...

**Comment by johnsensible** [comment]
Done: https://pypi.python.org/pypi/django-sendfile/0.3.11

**Pull request merged**